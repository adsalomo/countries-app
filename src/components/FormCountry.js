import React from 'react';
import PropTypes from 'prop-types';
import { useForm } from '../hooks/useForm';

export const FormCountry = ({ setCountries }) => {
    const [propertiesForm, handleOnChange, handleReset] = useForm({
        country: '',
        peoples: '',
    });

    const { country, peoples } = propertiesForm;


    const handleSave = (e) => {
        e.preventDefault();

        if (!country || country.length <= 0 || !peoples || peoples.length <= 0) {
            return;
        }

        console.log('Hola mundo', country, peoples);
        setCountries((countries) => [...countries, { country, peoples: Number(peoples) }]);
        handleReset();
    }


    return (
        <form onSubmit={(e) => handleSave(e, propertiesForm)}>
            <div className="form-group">
                <label>Pais</label>
                <input onChange={handleOnChange} value={country ? country : ''} name="country" className="form-control" type="text" placeholder="Escriba pais" autoComplete="off" />
            </div>
            <div className="form-group">
                <label>Habitantes</label>
                <input onChange={handleOnChange} value={peoples ? peoples : ''} name="peoples" className="form-control" type="number" placeholder="Escriba numero" autoComplete="off" />
            </div>
            <button className="btn btn-primary btn-block" type="submit">Nuevo pais</button>
        </form>
    )
}

FormCountry.propTypes = {
    setCountries: PropTypes.func.isRequired,
}


