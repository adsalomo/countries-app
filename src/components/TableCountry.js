import React from 'react';
import PropTypes from 'prop-types';

export const TableCountry = ({ countries }) => {
    console.log('TableCountry');

    return (
        <table className="table mt-3">
            <thead>
                <tr>
                    <th>#</th>
                    <th>Nombre</th>
                    <th>NÃºmero habitantes</th>
                </tr>
            </thead>
            <tbody>
                {
                    countries && countries.length > 0
                    && countries.map(({ country, peoples }, i) => {
                        return <tr key={i}>
                            <td>{i + 1}</td>
                            <td>{country}</td>
                            <td>{peoples}</td>
                        </tr>
                    })
                }
            </tbody>
        </table>
    )
}

TableCountry.propTypes = {
    countries: PropTypes.array.isRequired,
}
