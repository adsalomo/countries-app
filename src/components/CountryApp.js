import React, { useEffect, useState } from 'react';
import { getCountries } from '../helpers/getCountries';
import { FormCountry } from './FormCountry';

export const CountryApp = () => {
    const [countries, setCountries] = useState([]);

    useEffect(() => {
        getCountries().then(resp => {
            setCountries(resp);
        });
    }, []);


    return (
        <>
            <h1>Countries App</h1>
            <hr />

            <FormCountry
                setCountries={setCountries}
            />

            <table className="table mt-3">
                <thead>
                    <tr>
                        <th>#</th>
                        <th>Nombre</th>
                        <th>NÃºmero habitantes</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        countries.length > 0 && countries.map(({ country, peoples }, i) => {
                            return <tr key={i}>
                                <td>{i + 1}</td>
                                <td>{country}</td>
                                <td>{peoples}</td>
                            </tr>
                        })
                    }
                </tbody>
            </table>
        </>
    )
}
