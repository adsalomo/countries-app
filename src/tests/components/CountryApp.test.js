import { shallow } from 'enzyme'
import { CountryApp } from '../../components/CountryApp';

import '@testing-library/jest-dom';

describe('Prueba componente CountryApp', () => {

    const wrapper = shallow(<CountryApp />)

    test('Debe traer la foto del componente', () => {
        expect(wrapper).toMatchSnapshot();
    });


});
