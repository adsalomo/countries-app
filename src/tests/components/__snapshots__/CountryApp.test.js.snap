// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente CountryApp Debe traer la foto del componente 1`] = `
<Fragment>
  <h1>
    Countries App
  </h1>
  <hr />
  <FormCountry
    handleSave={[Function]}
  />
  <table
    className="table mt-3"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Nombre
        </th>
        <th>
          NÃºmero habitantes
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
